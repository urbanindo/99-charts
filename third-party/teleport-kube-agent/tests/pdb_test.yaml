suite: PodDisruptionBudget
templates:
  - pdb.yaml
tests:
  - it: should create a PDB when enabled in values (pdb.yaml)
    values:
      - ../.lint/pdb.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: PodDisruptionBudget
      - matchSnapshot: {}

  - it: sets PodDisruptionBudget labels when specified
    values:
      - ../.lint/pdb.yaml
      - ../.lint/extra-labels.yaml
    asserts:
      - equal:
          path: metadata.labels.app\.kubernetes\.io/name
          value: teleport-kube-agent
      - equal:
          path: metadata.labels.resource
          value: poddisruptionbudget
      - matchSnapshot: {}
