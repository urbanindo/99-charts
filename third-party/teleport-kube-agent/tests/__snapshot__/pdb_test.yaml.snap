sets PodDisruptionBudget labels when specified:
  1: |
    apiVersion: policy/v1beta1
    kind: PodDisruptionBudget
    metadata:
      labels:
        app: RELEASE-NAME
        app.kubernetes.io/name: teleport-kube-agent
        resource: poddisruptionbudget
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      minAvailable: 2
      selector:
        matchLabels:
          app: RELEASE-NAME
should create a PDB when enabled in values (pdb.yaml):
  1: |
    apiVersion: policy/v1beta1
    kind: PodDisruptionBudget
    metadata:
      labels:
        app: RELEASE-NAME
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      minAvailable: 2
      selector:
        matchLabels:
          app: RELEASE-NAME
